### PR TITLE
Stacki inventory

### PR DESCRIPTION
**Pull request by bbyhuy**
##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME
Stacki Dynamic Inventory Script

##### ANSIBLE VERSION

```
ansible 2.1.2.0
  config file = /home/vagrant/automata_modules/ansible.cfg
  configured module search path = Default w/o overrides
```

##### SUMMARY

Stacki inventory script to dynamically get node information, required stacki-pro

```
$ ./stacki.py --list
{
    "all": {
        "hosts": [
            "192.168.200.50", 
            "192.168.200.51"
        ]
    }, 
    "backends": {
        "hosts": [
            "192.168.200.51"
        ]
    }, 
    "_meta": {
        "hostvars": {
            "test-1": {
                "box": "default", 
                "installaction": "install", 
                "runaction": "os", 
                "interfaces": [
                    {
                        "network": "private", 
                        "default": true, 
                        "ip": "192.168.200.51", 
                        "vlan": null, 
                        "host": "test-1", 
                        "module": null, 
                        "mac": "08:00:27:C7:10:D1", 
                        "interface": "enp0s17", 
                        "options": null, 
                        "channel": null, 
                        "name": null
                    }, 
                    {
                        "network": null, 
                        "default": null, 
                        "ip": null, 
                        "vlan": null, 
                        "host": "test-1", 
                        "module": null, 
                        "mac": "08:00:27:43:4b:51", 
                        "interface": "enp0s8", 
                        "options": "dhcp", 
                        "channel": null, 
                        "name": null
                    }
                ], 
                "appliance": "backend", 
                "cpus": 1, 
                "rank": "0", 
                "ansible_host": "192.168.200.51", 
                "host": "test-1", 
                "rack": "0"
            }, 
            "stacki": {
                "box": "default", 
                "installaction": "install", 
                "runaction": "os", 
                "interfaces": [
                    {
                        "network": "private", 
                        "default": true, 
                        "ip": "192.168.200.50", 
                        "vlan": null, 
                        "host": "stacki", 
                        "module": null, 
                        "mac": "08:00:27:e3:b4:5d", 
                        "interface": "enp0s8", 
                        "options": null, 
                        "channel": null, 
                        "name": "stacki"
                    }
                ], 
                "appliance": "frontend", 
                "cpus": 1, 
                "rank": "0", 
                "ansible_host": "192.168.200.50", 
                "host": "stacki", 
                "rack": "0"
            }
        }
    }, 
    "frontends": {
        "hosts": [
            "192.168.200.50"
        ]
    }
}

```

### CODE REVIEW

**Comment by bbyhuy** [comment]
@thaumos 

@bsanders isn't appearing as a direct mention

**Comment by bcoca** [comment]
you have merges in your commit history, please clean that up

File: contrib/inventory/stacki.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+
+def stack_host_list(endpoint, header, client):
+
+    stack_r = client.post(endpoint, data=json.dumps({ "cmd": "list host"}),
+                          headers=header)
+    import ast
+
+    return json.loads(ast.literal_eval(stack_r.text))
+
```
Comment (bsanders) [inline contrib/inventory/stacki.py:109]: I'm not sure the need for pulling in 'ast' here?  `client.post` should be a requests response object, which in addition to having a `text` attribute, also has a `json()` method.  So stack_r.json() should give you what you want.

File: contrib/inventory/stacki.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+def stack_net_list(endpoint, header, client):
+
+    stack_r = client.post(endpoint, data=json.dumps({ "cmd": "list host interface"}),
+                          headers=header)
+    import ast
```
Comment (bsanders) [inline contrib/inventory/stacki.py:115]: See previous comment

File: contrib/inventory/stacki.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+
+def format_meta(hostdata, intfdata, config):
+    use_hostnames = config['use_hostnames']
+    meta = dict(all=dict(hosts=list()),
+                frontends=dict(hosts=list()),
+                backends=dict(hosts=list()),
+                _meta=dict(hostvars=dict()))
+    for host in hostdata:
+        del host['environment']
```
Comment (bsanders) [inline contrib/inventory/stacki.py:127]: Not sure it's safe to delete keys while iterating through a dictionary.  `for host in hostdata.keys()` is safer as it creates a new list to iterate over.

File: contrib/inventory/stacki.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+# Stacki inventory script
+# Configure stacki.yml with proper auth information and place in the following:
+#  - ../inventory/stacki.yml
+#  - /etc/stacki/stacki.yaml
```
Comment (bsanders) [inline contrib/inventory/stacki.py:21]: I might be missing some significance, but should these file extensions be the same?  .yaml vs .yml?

File: contrib/inventory/stacki.py [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+#  - /etc/ansible/stacki.yml
+# The stacki.yaml file can contain entries for authentication information
```
Comment (bsanders) [inline contrib/inventory/stacki.py:23]: See above

File: contrib/inventory/stacki.py [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+# regarding the Stacki front-end node.
+#
+# use_hostnames uses hostname rather than interface ip to as connection
```
Comment (bsanders) [inline contrib/inventory/stacki.py:26]: 'rather than interface ip **to** as connection'  Typo?

**Comment by bbyhuy** [comment]
closing and re-opening new PR, commit history too messy...

**Pull request closed without merging**